### PR TITLE
chore: bump `prettier` and `eslint-plugin-prettier`

### PR DESCRIPTION
**Pull request by mrazauskas**
Closes #20082
Closes #20179

`prettier` and `eslint-plugin-prettier` should be updated together. Since this is major version of `prettier`, there are few changes in formatting.

### CODE REVIEW

File: .prettierrc.yml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 tabWidth: 2
-trailingComma: all
```
Comment (mrazauskas) [inline .prettierrc.yml:2]: Shipped as new default: https://prettier.io/docs/en/options.html#trailing-commas

**Review by Jolg42 — APPROVED** [review]
Thanks @mrazauskas for the PR 💚

**Comment by mrazauskas** [comment]
@Jolg42 Sorry, I forgot one detail. Current, if Prettier v3 is installed, Jest fails to update inline snapshots. There are two solutions:

- downgrade Prettier to v2
- add `prettierPath: null` to Jest config. This will make sure Jest does not run Prettier, but new inline snapshots should be formatted outside Jest.

What would you prefer?

As you see in CI, all works with existing snaps. The problem accrues if they have to be updated or created.

**Comment by Jolg42** [comment]
Ouch 😢 

It sounds to me like reverting is easier/best until this is figured out then.

**Comment by Jolg42** [comment]
PR created from "Revert" UI button https://github.com/prisma/prisma/pull/20182

**Comment by mrazauskas** [comment]
> It sounds to me like reverting is easier/best until this is figured out then.

Agreed. I will come back after the issue is fixed on Jest side.

**Comment by Jolg42** [comment]
Thank you!

**Comment by Jolg42** [comment]
Now Jest will error in next version
https://github.com/jestjs/jest/pull/14367

Tracking issue to monitor
https://github.com/jestjs/jest/issues/14305

**Pull request merged**